### PR TITLE
fix: validate `ignorePatterns` constructor option in `FlatESLint` class

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Adds missing validation for `ignorePatterns` constructor option of the `FlatESLint` class.

#### What changes did you make? (Give an overview)

Updated `function processOptions` to throw a readable message when `ignorePatterns` is not a non-empty string, an array of non-empty strings, or `null`.

#### Is there anything you'd like reviewers to focus on?

I'm not sure if this option should be accepting `string` value as that could be misleading. `ignorePatterns: string` might be interpreted as if the string can contain multiple patterns (e.g., a newline-delimited list).

### CODE REVIEW

**Review by snitin315 — APPROVED** [review]
LGTM. I'll leave it open for others to review.

> I'm not sure if this option should be accepting string value as that could be misleading. ignorePatterns: string might be interpreted as if the string can contain multiple patterns (e.g., a newline-delimited list).

I believe the string type comes when using CLI `--ignore-pattern '!.build'`. Or does this also evaluates to `ignorePatterns: ['!.build']`?

**Review by nzakas — APPROVED** [review]
LGTM. Didn't realize we weren't validating this option.

**Comment by nzakas** [comment]
> I'm not sure if this option should be accepting string value as that could be misleading. ignorePatterns: string might be interpreted as if the string can contain multiple patterns (e.g., a newline-delimited list).

Oops, missed this. I think we can enforce that this option must take an array of strings and not a single string. Since the option is passed an array from the CLI, and `ESLint` also allows arrays in `ignorePatterns`, I think we're safe to limit to arrays.

**Comment by mdjermanovic** [comment]
Updated to disallow `ignorePatterns: string`.

**Review by nzakas — APPROVED** [review]
LGTM. Just waiting until after we're cleared of a patch release to merge.

**Review by fasttime — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**
